lib/OnyxUtils.ts

```diff
@@ -208,7 +216,7 @@ function get(key) {
 
     // When a value retrieving task for this key is still running hook to it
     if (cache.hasPendingTask(taskName)) {
-        return cache.getTaskPromise(taskName);
+        return cache.getTaskPromise(taskName) as Promise<OnyxValue<OnyxKey>>;
     }
```

Comment:
While I see that this won't cause an error because I checked the implementation of these functions, it again uses cast which I think signals not a good design in the code. What about just doing:

```
const taskPromise = cache.getTaskPromise(taskName);
if (taskPromise !== undefined) {
    return taskPromise;
}
```

Another approach is that `cache.getTaskPromise` should throw when it doesn't find the task, then `undefined` can be removed from the return type of `cache.getTaskPromise` and you can use `cache.hasPendingTask` like you do now.

Reply:
That's actually a really cool idea! What do you think about doing it as a follow-up PR to avoid bloating this PR any further?

Reply:
Sounds good to me!